src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/VirtualDocument.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    public abstract class VirtualDocument
```

Comment:
>VirtualDocument [](start = 26, length = 15)

Nit: It would be great to have some comments (doesn't have to be XML-doc) with brief explanation of the role of each class. I kind of remember it from our conversation yesterday, but I still think it would be good to have that in the code